substrate/frame/contracts-registry/src/lib.rs

```diff
@@ -284,7 +284,10 @@ pub mod pallet {
 
 			let contracts =
 				Contracts::<T, I>::get(contract_id).ok_or(Error::<T, I>::ContractNotFound)?;
-			ensure!(contracts.parties.contains(&who), Error::<T, I>::NotParty);
+
+			if let Some(parties) = contracts.parties {
+				ensure!(parties.contains(&who), Error::<T, I>::NotParty);
```

Comment:
We need to be prepared to handle thousands of signatures, we can't keep it in vec.